connect-location/src/main/java/com/ifttt/location/ConnectLocation.java

```diff
@@ -0,0 +1,97 @@
+package com.ifttt.location;
+
+import android.content.Context;
+import androidx.annotation.VisibleForTesting;
+import com.ifttt.connect.Connection;
+import com.ifttt.connect.ConnectionApiClient;
+import com.ifttt.connect.CredentialsProvider;
+import com.ifttt.connect.ErrorResponse;
+import com.ifttt.connect.Feature;
+import com.ifttt.connect.UserFeature;
+import com.ifttt.connect.UserFeatureField;
+import com.ifttt.connect.UserFeatureStep;
+import com.ifttt.connect.ui.ButtonStateChangeListener;
+import com.ifttt.connect.ui.ConnectButton;
+import com.ifttt.connect.ui.ConnectButtonState;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public final class ConnectLocation implements ButtonStateChangeListener {
+
+    private static ConnectLocation INSTANCE = null;
+    private AwarenessGeofenceProvider awarenessGeofenceProvider;
+    private ConnectionApiClient connectionApiClient;
+
+    final static String FIELD_TYPE_LOCATION_ENTER = "LOCATION_ENTER";
+    final static String FIELD_TYPE_LOCATION_EXIT = "LOCATION_EXIT";
+    final static String FIELD_TYPE_LOCATION_ENTER_EXIT =
+            "LOCATION_ENTER_OR_EXIT";
+
+    private final static List<String> locationFieldTypesList = Arrays.asList(FIELD_TYPE_LOCATION_ENTER, FIELD_TYPE_LOCATION_EXIT, FIELD_TYPE_LOCATION_ENTER_EXIT);
+
+    public static synchronized ConnectLocation init(Context context, ConnectionApiClient apiClient) {
+        if (INSTANCE == null) {
+            INSTANCE = new ConnectLocation(context);
+        }
+        INSTANCE.connectionApiClient = apiClient;
```

Comment:
There are two ways to get this client - 1. the activity sends us the client it wants to use. 2. we get a credenntial provider and have to instantiate a new client

If we want to abstract this client initialization, into the private constructor, we will have to pass these params to the constructor and do another check there, or have two separate private constructors. Since we already have two init methods specifically for differentiating how we initialize this class, I think the current approach is better.

Reply:
hmm we could just let the private constructor take in a client. Let's go with that.

Reply:
>hmm we could just let the private constructor take in a client. 

That was what I meant, because we always create an instance and then assign the apiClient right away.